src/sdk/account/toNexusAccount.ts

```diff
@@ -562,3 +562,49 @@ export const toNexusAccount = async (
     }
   })
 }
+
+/**
+ * @description Computes the counterfactual address for a Nexus account without creating it
+ * @param parameters - Configuration parameters
+ * @param parameters.chain - The chain netwrk
+ * @param parameters.transport - The transport configuration
+ * @param parameters.signerAddress - The EOA address that will control the account
+ * @param parameters.index - Optional index for the account (defaults to 0n)
+ * @param parameters.factoryAddress - Optional factory address
+ * @returns The computed counterfactual address
+ */
+export const computeNexusAddress = async ({
+  chain,
+  transport,
+  signerAddress,
+  index = 0n,
+  factoryAddress = k1ValidatorFactoryAddress
+}: {
+  chain: Chain
+  transport: ClientConfig["transport"]
+  signerAddress: Address
+  index?: bigint
+  factoryAddress?: Address
+}): Promise<Address> => {
+  const publicClient = createPublicClient({
+    chain,
+    transport
+  })
+
+  try {
+    const address = (await publicClient.readContract({
```

Comment:
Check the other open PR. I think I prefer the `getCounterFactual` way of getting an address because it relies on the entrypoint (which will always be deployed) instead of the factory (which may not necessarily have been deployed)

Reply:
Is there any other PR with `getCounterFactual` method or should I create one?

Reply:
No we don't need one. Gabi has effectively removed the current logic that sits behind `getAddress`, replaced it with the logic that was in `getCounterFactualAddress`, and then made them an alias of one another